lib/betterdocs/generator/swagger.rb

```diff
@@ -0,0 +1,391 @@
+        create_swagger(map, dirname)
+        # create_assets
+        self
+      end
+
+      def configure_for_creation
+        infobar.puts color(40, "Setting asset_host to #{Betterdocs::Global.asset_host.inspect}.")
+        Betterdocs.rails.configuration.action_controller.asset_host = Betterdocs::Global.asset_host
+        options = {
+          host: Betterdocs::Global.api_host,
+          protocol: Betterdocs::Global.api_protocol
+        }
+        infobar.puts color(40, "Setting default_url_options to #{options.inspect}.")
+        Betterdocs.rails.application.routes.default_url_options = options
+        self
+      end
+
+      def add_global_info(map)
+        slugs = get_request_url_slugs(sections.values.first.first.request)
+        map[:info] = { title: project_name, version: slugs[:ver] }
+        map[:servers] =
+          [{ url: "#{slugs[:protocol]}://#{[slugs[:host], slugs[:lang], "api_#{slugs[:ver]}"].join('/')}" }]
+      end
+
+      def create_sections(map)
+        Infobar(total: sections.size)
+        sections.values.each do |section|
+          infobar.progress(
+            message: " Section #{section.name.to_s.inspect} %c/%t in %te ETA %e @%E ",
+            force: true
+          )
+          @only =~ section.name or next if @only
+
+          map = get_section_data(map, section)
+        end
+        infobar.finish message: ' %t sections created in %te, completed @%E '
+        infobar.newline
+        self
+      end
+
+      def create_swagger(map, dirname)
+        name = 'swagger.json'
+        cd dirname do
+          infobar.puts color(40, 'Creating swagger.')
+          json = JSON.pretty_generate(map)
+          render_to(name, json)
+        end
+        self
+      end
+
+      def fail_while_rendering(exception)
+        message = color(
+          231,
+          on_color(124, " *** ERROR #{exception.class}: #{exception.message.inspect} ***")
+        )
+        infobar.puts message
+        exit 1
+      end
+
+      def add_param(hash, name, param_in, required = false, type = nil, description = '', schema = nil)
+        hash[:parameters] = [] unless hash.key?(:parameters)
+        p = { in: param_in, required: required, description: description, name: name }
+        p[:schema] = { type: type } unless type.nil?
+        p[:schema] = schema unless schema.nil?
+        hash[:parameters].push(p)
+        hash
+      end
+
+      def add_default_list_params(hash)
+        p = add_param(hash, 'page', 'query', false, 'integer')
+        add_param(p, 'per_page', 'query', false, 'integer')
+      end
+
+      def get_list_response_wrapper_schema
+        { type: 'object', properties: {
+          total_entries: { type: 'integer' },
+          offset: { type: 'integer' },
+          total_pages: { type: 'integer' },
+          current_page: { type: 'integer' },
+          per_page: { type: 'integer' }
+        }, required: %w[total_entries offset current_page per_page total_pages data] }
```

Comment:
```suggestion
        { 
          type: 'object',
          properties: {
            total_entries: { type: 'integer' },
            offset: { type: 'integer' },
            total_pages: { type: 'integer' },
            current_page: { type: 'integer' },
            per_page: { type: 'integer' }
          },
          required: %w[total_entries offset current_page per_page total_pages data]
        }
```

more standard formatting